packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraSessionPresetsTest.m

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera_avfoundation;
+@import camera_avfoundation.Test;
+@import AVFoundation;
+@import XCTest;
+#import <OCMock/OCMock.h>
+#import "CameraTestUtils.h"
+
+/// Includes test cases related to photo capture operations for FLTCam class.
+@interface FLTCamSessionPresetsTest : XCTestCase
+
+@end
+
+@implementation FLTCamSessionPresetsTest
+
+- (void)testResolutionPresetWithCanSetSessionPresetMax_mustUpdateCaptureSessionPreset {
+  NSString *expectedUltraHighPreset = AVCaptureSessionPreset3840x2160;
+  NSString *expectedMaxPreset = AVCaptureSessionPresetInputPriority;
+
+  id videoSessionMock = OCMClassMock([AVCaptureSession class]);
+  OCMStub([videoSessionMock addInputWithNoConnections:[OCMArg any]]);  // no-op
+  OCMStub([videoSessionMock canSetSessionPreset:[OCMArg any]]).andReturn(YES);
+  OCMExpect([videoSessionMock setSessionPreset:[OCMArg checkWithBlock:^BOOL(id value) {
+                                // Return YES if the value is one of the expected presets
+                                return [value isEqualToString:expectedUltraHighPreset] ||
+                                       [value isEqualToString:expectedMaxPreset];
```

Comment:
can you explicitly mock the values and verify each of them?

Reply:
Thanks for the review! 
I'm not sure I understood your message here. If you are suggesting two different tests (one for `expectedUltraHighPreset` and one for `expectedMaxPreset`), then that's not really the expected behaviour. 
The test is for the case where camera receives `FLTResolutionPresetMax`, and in that case `sessionPreset` can be one of these values and both are acceptable.
If I missed your point - could you please clarify what you meant?

Reply:
>  in that case sessionPreset can be one of these two values and both are acceptable.

Can you explain when will it be one value and when the other value? It's unclear which one will be true when running the test.

Reply:
If you take a look in the file [FLTCam.m](https://github.com/flutter/packages/pull/5245/files#diff-1af831ad95b26582a2afa02afb68f6de8596c2d118f39d246523cf0a25f95aa4) at line 372, there are two options on how to handle FLTResolutionPresetMax. Either we can find `bestFormat`, in which case `sessionPreset` is set to `AVCaptureSessionPresetInputPriority`, or the `bestFormat` is nil, in which case we try to set `sessionPreset` to `AVCaptureSessionPreset3840x2160`. 

Since method `canSetSessionPreset` of our mock object always returns true, we can expect that if we try to set `AVCaptureSessionPreset3840x2160`, we will always succeed. That's how we get only to possible outcomes. 


> It's unclear which one will be true when running the test.
About that - we could mock further and provide our videoSessionMock with availableFormats, split the test in two different tests - one that covers case with the best format and one that covers case with no formats and `AVCaptureSessionPreset3840x2160`. But since both of those cases are an acceptable response to setting FLTResolutionPresetMax, I didn't see the need to go this far

Reply:
> Since method `canSetSessionPreset` of our mock object always returns true, we can expect that if we try to set `AVCaptureSessionPreset3840x2160`, we will always succeed. That's how we get only to possible outcomes.

If we have two codepaths, we should be testing both of them.

> But since both of those cases are an acceptable response to setting FLTResolutionPresetMax (as I explained above), I didn't see the need to go this far

Why would we want no test coverage that one of the possible outcomes actually works?

Reply:
> If we have two codepaths, we should be testing both of them.

Thank you! I'll add another test


> Why would we want no test coverage that one of the possible outcomes actually works?

I'm not sure, since this whole behavior of resolution presets wasn't covered by any test at all :)

Reply:
https://github.com/flutter/flutter/wiki/Plugin-Tests#faq-do-i-need-to-add-tests-even-if-the-part-of-the-code-im-changing-isnt-already-tested

Reply:
https://github.com/flutter/flutter/wiki/Plugin-Tests#faq-do-i-need-to-add-tests-even-if-the-part-of-the-code-im-changing-isnt-already-tested